python/pyiceberg/catalog/hive.py

```diff
@@ -159,16 +156,56 @@ def _construct_parameters(metadata_location: str, previous_metadata_location: Op
 def _annotate_namespace(database: HiveDatabase, properties: Properties) -> HiveDatabase:
     params = {}
     for key, value in properties.items():
-        if key == "comment":
+        if key == COMMENT:
             database.description = value
-        elif key == "location":
+        elif key == LOCATION:
             database.locationUri = value
         else:
             params[key] = value
     database.parameters = params
     return database
 
 
+hive_primitive_types = {
+    BooleanType: "boolean",
+    IntegerType: "int",
+    LongType: "bigint",
+    FloatType: "float",
+    DoubleType: "double",
+    DateType: "date",
+    TimeType: "string",
+    TimestampType: "timestamp",
+    StringType: "string",
+    UUIDType: "string",
+    BinaryType: "binary",
+    FixedType: "binary",
+}
+
+
+class HiveSchemaConstructor(SchemaVisitor[str]):
+    def schema(self, schema: Schema, struct_result: str) -> str:
+        return struct_result
+
+    def struct(self, struct: StructType, field_results: List[str]) -> str:
+        return f"struct<{', '.join(field_results)}>"
+
+    def field(self, field: NestedField, field_result: str) -> str:
+        return f"{field.name}: {field_result}"
+
+    def list(self, list_type: ListType, element_result: str) -> str:
+        return f"array<{element_result}>"
+
+    def map(self, map_type: MapType, key_result: str, value_result: str) -> str:
+        # Key has to be primitive for Hive
+        return f"map<{key_result}, {value_result}>"
+
+    def primitive(self, primitive: PrimitiveType) -> str:
+        if isinstance(primitive, DecimalType):
+            return f"DECIMAL({primitive.precision}, {primitive.scale})"
```

Comment:
Why is decimal upper case but the others aren't?

Reply:
I copied this from the docs, but they are very inconsistent. I tested, and the uppercase one actually fails when reading.